### PR TITLE
use only the wrapper in the case of a collapsible table

### PR DESCRIPTION
**Pull request by pochretien**
fix the tbody tag for the simple table

### CODE REVIEW

File: src/components/tables/table-children/TableChildBody.tsx [hunk 1]
```diff
@@ -58,25 +58,27 @@ export const TableChildBody = (props: ITableChildBodyProps): JSX.Element => {
...
+        }>
+        {tableHeadingRowContent}
+    </TableHeadingRowConnected>;
+
+    return collapsibleRow
+        ? <TableCollapsibleRowWrapper className={tableRowWrapperClasses}
+            collapsibleRow={collapsibleRow}>
+            {tableHeadingRowConnectedNode}
+        </TableCollapsibleRowWrapper>
+        : tableHeadingRowConnectedNode;
```
Comment (MichelML) [inline src/components/tables/table-children/TableChildBody.tsx]: i think you could just rewrite the end of this Component as such:  
  
```

    const tableHeadingRowConnectedNode = (
        <TableHeadingRowConnected
            id={headingAndCollapsibleId}
            tableId={props.tableId}
            className={tableRowClasses}
            isCollapsible={!!collapsibleData}
            onClickCallback={() => {
                if (props.onRowClick) {
                    props.onRowClick(props.getActions(props.rowData));
                }
            }}
            onDoubleClick={() => props.getActions(props.rowData)
                .filter((action) => action.callOnDoubleClick)
                .forEach((action) => action.trigger())
            }>
            {tableHeadingRowContent}
        </TableHeadingRowConnected>
    );

    return collapsibleRow
        ? (
            <TableRowWrapper className={tableRowWrapperClasses}>
                {tableHeadingRowConnectedNode}
                {collapsibleRow}
            </TableRowWrapper>
        )
        : tableHeadingRowConnectedNode;  
```  
I don't think it is necessary to change the name and structure of TableRowWrapper.tsx imho, and you'll have less UTs to fix :D (you currently have 6)

File: src/components/tables/Table.tsx [hunk 2]
```diff
@@ -210,5 +210,9 @@ export class Table extends React.Component<ITableProps, {}> {
                 />
             );
         });
+
+        return this.props.collapsibleFormatter
+            ? tableBodyNode
+            : <tbody>{tableBodyNode}</tbody>;
```
Comment (MichelML) [inline src/components/tables/Table.tsx]: I would suggest adding a comment here to explain the situation since it is a peculiar case

**Comment by MichelML** [comment]
muy bueno idea for the fix by the way 👍 👍 👍 👍 😄 🎉

File: src/components/tables/TableRowWrapper.tsx [hunk 3]
```diff
@@ -1,4 +1,13 @@
...
+export interface ITableCollapsibleRowWrapperProps {
+    collapsibleRow: React.ReactNode;
+    children: React.ReactNode;
+    className?: string;
+}
+
+export const TableCollapsibleRowWrapper = (props?: ITableCollapsibleRowWrapperProps): JSX.Element =>
+    <tbody className={props.className}>
+        {props.children}
+        {props.collapsibleRow}
```
Comment (sroy3) [inline src/components/tables/TableRowWrapper.tsx]: Can't the `collapsibleRow` be just another `children` instead?

File: src/components/tables/TableRowWrapper.tsx [hunk 4]
```diff
@@ -1,4 +1,13 @@
...
 
-export const TableRowWrapper = (props?: any): JSX.Element =>
-    <tbody className={props.className}>{props.children}</tbody>;
+export interface ITableCollapsibleRowWrapperProps {
+    collapsibleRow: React.ReactNode;
+    children: React.ReactNode;
+    className?: string;
+}
+
+export const TableCollapsibleRowWrapper = (props?: ITableCollapsibleRowWrapperProps): JSX.Element =>
```
Comment (sroy3) [inline src/components/tables/TableRowWrapper.tsx]: Might want to change the name of the file as well to be `TableCollapsibleRowWrapper`.

File: src/components/tables/TableRowWrapper.tsx (hunk 3 above)
Comment (pochretien) [reply src/components/tables/TableRowWrapper.tsx]: fixed

File: src/components/tables/Table.tsx [hunk 5]
```diff
@@ -157,23 +162,39 @@ export class Table extends React.Component<ITableProps, {}> {
...
         );
     }
 
+    private setFixedHeaderWrapper(tableElement: React.ReactNode) {
+        return this.props.withFixedHeader
+            ? <div className='fixed-header-table-container'>
+                <div className='fixed-header-table'>
+                    {tableElement}
+                </div>
+            </div>
```
Comment (MichelML) [inline src/components/tables/Table.tsx]: use parentheses https://github.com/coveo/tsjs/tree/master/react_redux#parentheses

File: src/components/tables/Table.tsx [hunk 6]
```diff
@@ -207,8 +228,15 @@ export class Table extends React.Component<ITableProps, {}> {
...
                     collapsibleFormatter={this.props.collapsibleFormatter}
                     onRowClick={(actions: IActionOptions[]) => this.props.onRowClick(actions)}
+                    handleOnRowClick={this.props.handleOnRowClick}
                 />
             );
         });
+
+        return this.props.collapsibleFormatter
+            ? tableBodyNode
+            : <tbody className={classNames(this.props.tableBodyClasses)}>
```
Comment (MichelML) [inline src/components/tables/Table.tsx]: use parentheses https://github.com/coveo/tsjs/tree/master/react_redux#parentheses

File: src/components/tables/TableHeaderCell.tsx [hunk 7]
```diff
@@ -43,6 +44,13 @@ export class TableHeaderCell extends React.Component<ITableHeaderCellProps, any>
         }
     }
 
+    private setWithFixedHeaderWrapper(titleNode: React.ReactNode, sortIcon: React.ReactNode) {
+        return <div className='fixed-header-container'>
```
Comment (MichelML) [inline src/components/tables/TableHeaderCell.tsx]: use parentheses

File: src/components/tables/table-children/TableChildBody.tsx [hunk 8]
```diff
@@ -58,25 +58,31 @@ export const TableChildBody = (props: ITableChildBodyProps): JSX.Element => {
...
-                        props.onRowClick(props.getActions(props.rowData));
-                    }
-                }}
-                onDoubleClick={() => props.getActions(props.rowData)
-                    .filter((action) => action.callOnDoubleClick)
-                    .forEach((action) => action.trigger())
-                }>
-                {tableHeadingRowContent}
-            </TableHeadingRowConnected>
+    const tableHeadingRowConnectedNode = <TableHeadingRowConnected
```
Comment (MichelML) [inline src/components/tables/table-children/TableChildBody.tsx]: use parentheses https://github.com/coveo/tsjs/tree/master/react_redux#parentheses

**Pull request merged**